Library/ViewModels/SettingsNewslettersCellViewModel.swift

```diff
@@ -0,0 +1,162 @@
+
+    let initialUser = self.initialUserProperty.signal.skipNil()
+
+    let newsletterOn: Signal<(Newsletter, Bool), NoError> = newsletter
+      .takePairWhen(self.newslettersSwitchTappedProperty.signal.skipNil())
+      .map { newsletter, isOn in (newsletter, isOn) }
+
+    self.showOptInPrompt = newsletterOn
+      .filter { _, on in AppEnvironment.current.config?.countryCode == "DE" && on }
+      .map { newsletter, _ in newsletter.displayableName }
+
+    let userAttributeChanged = newsletter
+      .takePairWhen(self.newslettersSwitchTappedProperty.signal.skipNil())
+      .map { newsletter, isOn in
+        (UserAttribute.newsletter(newsletter), isOn)
+      }.logEvents(identifier: "user attribute changed")
+
+    let updatedUser = initialUser
+      .switchMap { user in
+        userAttributeChanged.scan(user) { user, attributeAndOn in
+          let (attribute, on) = attributeAndOn
+          return user |> attribute.lens .~ on
+        }
+    }.logEvents(identifier: "updated user")
+
+    let updateUserAllOn = initialUser
+      .takePairWhen(self.allNewslettersSwitchProperty.signal.skipNil())
+      .map { user, on in
+        return user
+          |> User.lens.newsletters .~ User.NewsletterSubscriptions.all(on: on)
+    }.logEvents(identifier: "updated all")
+
+    let updateEvent = Signal.merge(updatedUser, updateUserAllOn)
+      .switchMap { user in
+        AppEnvironment.current.apiService.updateUserSelf(user)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+    }
+    .logEvents(identifier: "update event")
+
+    self.unableToSaveError = updateEvent.errors()
+      .map { env in
+        env.errorMessages.first ?? Strings.profile_settings_error()
+      }.logEvents(identifier: "Couldn't update user")
+
+    let initialUserOnError = initialUser
+      .takeWhen(self.unableToSaveError)
+      .map { previous in previous }
+
+    self.updateCurrentUser = Signal.merge(initialUser,
+                                          updatedUser,
+                                          updateUserAllOn,
+                                          initialUserOnError)
+      .takeWhen(updateEvent.values().ignoreValues())
+
+    self.subscribeToAllSwitchIsOn = initialUser
+      .map(userIsSubscribedToAll(user:))
+
+    self.switchIsOn = initialUser
+      .combineLatest(with: newsletter)
+      .map(userIsSubscribed(user:newsletter:)).logEvents(identifier: "switchIsOn emitted")
+
+    // Koala
+    userAttributeChanged
+      .observeValues { attribute, on in
+        switch attribute {
+        case let .newsletter(newsletter):
+          AppEnvironment.current.koala.trackChangeNewsletter(
+            newsletterType: newsletter, sendNewsletter: on, project: nil, context: .settings
+          )
+        default: break
+      }
+    }
+  }
+
+  fileprivate let awakeFromNibProperty = MutableProperty(())
+  public func awakeFromNib() {
```

Comment:
I think you can remove this since I don't see any reference to `awakeFromNibProperty`

Reply:
Oh yeah, I forgot to delete that.